{"context": " More than 1 year has passed since last update.C#.NET\u3000\u3067\u3001try ~ catch \u3067\u6355\u6349\u3055\u308c\u306a\u3044\u4f8b\u5916\u3092\u88dc\u8db3\u3057\u305f\u3044\u5834\u5408\u3001\nAppDomain.UnhandledException \u30a4\u30d9\u30f3\u30c8\u30cf\u30f3\u30c9\u30e9\u3092\u8a2d\u5b9a\u3057\u3066\u304a\u3051\u3070\u3001\u305d\u3053\u3067\u6355\u6349\u3057\u3066\u304f\u308c\u307e\u3059\u3002\n\u3053\u308c\u3067 try \uff5e catch \u3067\u6355\u6349\u3057\u5fd8\u308c\u3066\u3082\u3001\u5b89\u5fc3\u30a2\u30f3\u30b3\u30fc\u30eb\u30ef\u30c3\u30c8\u306d\uff01\nForm\u30a2\u30d7\u30ea\u30b1\u30fc\u30b7\u30e7\u30f3\u3001\u30b3\u30f3\u30bd\u30fc\u30eb\u30a2\u30d7\u30ea\u30b1\u30fc\u30b7\u30e7\u30f3\u3001\u4e21\u65b9\u3067\u4f7f\u3048\u307e\u3059\u3002\n\n\u30b5\u30f3\u30d7\u30eb\u30b3\u30fc\u30c9\n\nprogram.cs\n/// <summary>\n/// try ~ catch \u3067\u6355\u6349\u3055\u308c\u3066\u306a\u3044\u4f8b\u5916\u3092UnhandledException\u3067\u6355\u6349\u3059\u308b\u30b5\u30f3\u30d7\u30eb\n/// </summary>\nusing System;\nusing System.IO;\nusing System.Text;\nusing System.Windows.Forms;\n\nnamespace SampleUnhandledException\n{\n    class Program\n    {\n        public static void Main(string[] args)\n        {\n            AppDomain.CurrentDomain.UnhandledException += new UnhandledExceptionEventHandler(CurrentDomain_UnhandledException);\n\n            //try ~ catch \u3067\u6355\u6349\u3055\u308c\u3066\u308b\u4f8b\u5916\n            try {\n                throw new Exception(\"1\");\n            } catch (Exception e) {\n                Console.WriteLine(\"Catch clause caught : \" + e.Message);\n            }\n\n            //try ~ catch \u3067\u6355\u6349\u3055\u308c\u3066\u306a\u3044\u4f8b\u5916\n            throw new Exception(\"2\");\n        }\n\n        /// <summary>\n        /// try ~ catch \u3067\u6355\u6349\u3055\u308c\u3066\u306a\u3044\u4f8b\u5916\u3092\u51e6\u7406\u3059\u308b\u30a4\u30d9\u30f3\u30c8\u30cf\u30f3\u30c9\u30e9\n        /// </summary>\n        /// <param name=\"sender\">\u9001\u4fe1\u8005</param>\n        /// <param name=\"args\">\u4f8b\u5916\u30aa\u30d6\u30b8\u30a7\u30af\u30c8</param>\n        static void CurrentDomain_UnhandledException(object sender, UnhandledExceptionEventArgs args)\n        {\n            try\n            {\n                Exception ex = (Exception)args.ExceptionObject;\n\n                //\u30a8\u30e9\u30fc\u51e6\u7406\n                MessageBox.Show(GetExceptionInfoString(ex));\n            }\n            finally\n            {\n                Application.Exit(); // \u660e\u793a\u7684\u306b\u7d42\u4e86\u3055\u305b\u308b\u5fc5\u8981\u304c\u3042\u308b\n            }\n        }\n\n        /// <summary>\n        /// \u4f8b\u5916\u30aa\u30d6\u30b8\u30a7\u30af\u30c8\u304b\u3089\u30c7\u30d0\u30c3\u30b0\u306b\u5fc5\u8981\u306a\u60c5\u5831\u3092\u6587\u5b57\u5217\u3067\u8fd4\u3059\n        /// </summary>\n        /// <param name=\"ex\">\u4f8b\u5916\u30aa\u30d6\u30b8\u30a7\u30af\u30c8</param>\n        /// <returns>\u4f8b\u5916\u60c5\u5831\u306e\u6587\u5b57\u5217</returns>\n        static string GetExceptionInfoString(Exception ex)\n        {\n            var sb = new StringBuilder();\n\n            sb.AppendFormat(\"Message={0}\\n\", ex.Message);\n            sb.AppendFormat(\"Source={0}\\n\", ex.Source);\n            sb.AppendFormat(\"HelpLink={0}\\n\", ex.HelpLink);\n            sb.AppendFormat(\"TargetSite={0}\\n\", ex.TargetSite.ToString());\n            sb.AppendFormat(\"StackTrace={0}\\n\", ex.StackTrace);\n\n            return sb.ToString();\n        }\n\n    }\n}\n\n\n\n\u5b9f\u884c\u7d50\u679c\n\nLog4Net \u3068\u7d44\u307f\u5408\u308f\u305b\u308b\u3068\u3088\u3055\u3052\u304b\u3082\u3057\u308c\u307e\u305b\u3093\u306d\u3002\n\n\u53c2\u8003\n\nAppDomain.UnhandledException \u30a4\u30d9\u30f3\u30c8 (System)\n\nC#.NET\u3000\u3067\u3001try ~ catch \u3067\u6355\u6349\u3055\u308c\u306a\u3044\u4f8b\u5916\u3092\u88dc\u8db3\u3057\u305f\u3044\u5834\u5408\u3001\nAppDomain.UnhandledException \u30a4\u30d9\u30f3\u30c8\u30cf\u30f3\u30c9\u30e9\u3092\u8a2d\u5b9a\u3057\u3066\u304a\u3051\u3070\u3001\u305d\u3053\u3067\u6355\u6349\u3057\u3066\u304f\u308c\u307e\u3059\u3002\n\u3053\u308c\u3067 try \uff5e catch \u3067\u6355\u6349\u3057\u5fd8\u308c\u3066\u3082\u3001\u5b89\u5fc3\u30a2\u30f3\u30b3\u30fc\u30eb\u30ef\u30c3\u30c8\u306d\uff01\n\nForm\u30a2\u30d7\u30ea\u30b1\u30fc\u30b7\u30e7\u30f3\u3001\u30b3\u30f3\u30bd\u30fc\u30eb\u30a2\u30d7\u30ea\u30b1\u30fc\u30b7\u30e7\u30f3\u3001\u4e21\u65b9\u3067\u4f7f\u3048\u307e\u3059\u3002\n\n# \u30b5\u30f3\u30d7\u30eb\u30b3\u30fc\u30c9\n\n```csharp:program.cs\n/// <summary>\n/// try ~ catch \u3067\u6355\u6349\u3055\u308c\u3066\u306a\u3044\u4f8b\u5916\u3092UnhandledException\u3067\u6355\u6349\u3059\u308b\u30b5\u30f3\u30d7\u30eb\n/// </summary>\nusing System;\nusing System.IO;\nusing System.Text;\nusing System.Windows.Forms;\n\nnamespace SampleUnhandledException\n{\n\tclass Program\n\t{\n\t\tpublic static void Main(string[] args)\n\t\t{\n\t\t\tAppDomain.CurrentDomain.UnhandledException += new UnhandledExceptionEventHandler(CurrentDomain_UnhandledException);\n\t\t\t\n\t\t\t//try ~ catch \u3067\u6355\u6349\u3055\u308c\u3066\u308b\u4f8b\u5916\n\t\t\ttry {\n\t\t\t\tthrow new Exception(\"1\");\n\t\t\t} catch (Exception e) {\n\t\t\t\tConsole.WriteLine(\"Catch clause caught : \" + e.Message);\n\t\t\t}\n\n\t\t\t//try ~ catch \u3067\u6355\u6349\u3055\u308c\u3066\u306a\u3044\u4f8b\u5916\n\t\t\tthrow new Exception(\"2\");\n\t\t}\n\n\t\t/// <summary>\n\t\t/// try ~ catch \u3067\u6355\u6349\u3055\u308c\u3066\u306a\u3044\u4f8b\u5916\u3092\u51e6\u7406\u3059\u308b\u30a4\u30d9\u30f3\u30c8\u30cf\u30f3\u30c9\u30e9\n\t\t/// </summary>\n\t\t/// <param name=\"sender\">\u9001\u4fe1\u8005</param>\n\t\t/// <param name=\"args\">\u4f8b\u5916\u30aa\u30d6\u30b8\u30a7\u30af\u30c8</param>\n\t\tstatic void CurrentDomain_UnhandledException(object sender, UnhandledExceptionEventArgs args)\n\t\t{\n\t\t\ttry\n\t\t\t{\n\t\t\t\tException ex = (Exception)args.ExceptionObject;\n\t\t\t\t\n\t\t\t\t//\u30a8\u30e9\u30fc\u51e6\u7406\n\t\t\t\tMessageBox.Show(GetExceptionInfoString(ex));\n\t\t\t}\n\t\t\tfinally\n\t\t\t{\n\t\t\t\tApplication.Exit(); // \u660e\u793a\u7684\u306b\u7d42\u4e86\u3055\u305b\u308b\u5fc5\u8981\u304c\u3042\u308b\n\t\t\t}\n\t\t}\n\t\t\n\t\t/// <summary>\n\t\t/// \u4f8b\u5916\u30aa\u30d6\u30b8\u30a7\u30af\u30c8\u304b\u3089\u30c7\u30d0\u30c3\u30b0\u306b\u5fc5\u8981\u306a\u60c5\u5831\u3092\u6587\u5b57\u5217\u3067\u8fd4\u3059\n\t\t/// </summary>\n\t\t/// <param name=\"ex\">\u4f8b\u5916\u30aa\u30d6\u30b8\u30a7\u30af\u30c8</param>\n\t\t/// <returns>\u4f8b\u5916\u60c5\u5831\u306e\u6587\u5b57\u5217</returns>\n\t\tstatic string GetExceptionInfoString(Exception ex)\n\t\t{\n\t\t\tvar sb = new StringBuilder();\n\n\t\t\tsb.AppendFormat(\"Message={0}\\n\", ex.Message);\n\t\t\tsb.AppendFormat(\"Source={0}\\n\", ex.Source);\n\t\t\tsb.AppendFormat(\"HelpLink={0}\\n\", ex.HelpLink);\n\t\t\tsb.AppendFormat(\"TargetSite={0}\\n\", ex.TargetSite.ToString());\n\t\t\tsb.AppendFormat(\"StackTrace={0}\\n\", ex.StackTrace);\n\n\t\t\treturn sb.ToString();\n\t\t}\n\t\t\n\t}\n}\n```\n\n# \u5b9f\u884c\u7d50\u679c\n\n![\u5b9f\u884c\u7d50\u679c](https://lh6.googleusercontent.com/-0X0OmNDW564/UBqXwLHjQ6I/AAAAAAAAEYQ/LfApdDkqSD0/s640/2012-08-03_00h05_42.png)\n\nLog4Net \u3068\u7d44\u307f\u5408\u308f\u305b\u308b\u3068\u3088\u3055\u3052\u304b\u3082\u3057\u308c\u307e\u305b\u3093\u306d\u3002\n\n# \u53c2\u8003\n\n* [AppDomain.UnhandledException \u30a4\u30d9\u30f3\u30c8 (System)](http://msdn.microsoft.com/ja-jp/library/system.appdomain.unhandledexception.aspx)\n", "tags": ["C#", ".NET"]}